### ISSUE TITLE
Training with surface normal / sdf gradient supervision

### ISSUE THREAD

**Issue by LeiYangJustin** [nv-tlabs/nglod#36]
From the paper it seems the models are trained without surface normal supervision or sdf gradient supervision.

I would like to know if nglod can support these supervision as they are widely used in recent works on neural implicit modeling.

Regards

**Comment by joeylitalien** [comment]
Hi @LeiYangJustin,

You can always train our model with additionnal supervision. In the supplementary material of the paper, we justify not using Eikonal regularization to be able to fit non-metric SDFs, but this can be relaxed and added to the loss function if you want. For 3D supervision of normals, you simply have to compute normalized SDF gradients with respect to coordinate inputs on the grid, which is supported by Kaolin.

Indeed, we recently released [Kaolin Wisp](https://github.com/NVIDIAGameWorks/kaolin-wisp), which includes an improved implementation of NGLOD. In particular, you will find that `RenderBuffer` lets you pass additional channels from the renderer to the training loop. You can use this structure to easily supervise surface normals in image-space, if that's also something you're also interested in.

Hope this helps,
Joey
